Maximum recursion depth exceeded error when latex-printing 1/(5/6)

`Maximum recursion depth exceeded` error when latex-printing `1/(2/3)`

Code to reproduce:

```python
import sympy
s = sympy.Pow(sympy.Rational(2, 3), -1, evaluate=False)
print(sympy.latex(s))
```

Fails on versions 1.12rc1+, works on 1.11.1.
Git bisection gives commit 

```
commit e76564ad53a297079b0c567c607e1e57a9e75de6
Author: Chris Smith <smichr@gmail.com>
Date:   Fri Aug 12 19:53:10 2022 -0500

    23918: canonical Rational**x

 sympy/core/power.py                   | 12 +++++++-----
...
```

with the following [change](https://github.com/sympy/sympy/commit/e76564ad53a297079b0c567c607e1e57a9e75de6#diff-e5505839b03d72fa2630fe48bd505a715973e215cebe8ffc5eccd2efeebc5f17L985) in function `as_base_exp`:

```python
        b, e = self.args
        if b.is_Rational and b.p == 1 and b.q != 1:
            return Integer(b.q), -e
        return b, e
# vvvvvvv
        b, e = self.args
        if b.is_Rational and b.p < b.q and b.p > 0:  # <- problem here
            return 1/b, -e
        return b, e
```

This change cause an infinite loop:
1. `sympy/printing/latex.py` [calls](https://github.com/sympy/sympy/blob/1a7afca81a3b4bcf5c60a41f04c27eb677309d6d/sympy/printing/latex.py#L583)

```python
        numer, denom = fraction(expr, exact=True)
```

2. Inside fraction func (`sympy/simplify/radsimp.py` ) there is [fragment](https://github.com/sympy/sympy/blob/1a7afca81a3b4bcf5c60a41f04c27eb677309d6d/sympy/simplify/radsimp.py#L1112)

```python
...
            b, ex = term.as_base_exp()
            if ex.is_negative:
                if ex is S.NegativeOne:
                    denom.append(b)
                elif exact:
                    if ex.is_constant():
                        denom.append(Pow(b, -ex))
                    else:
                        numer.append(term)
                else:
                    denom.append(Pow(b, -ex))
            elif ex.is_positive:  # <- problem here
                numer.append(term)
...
```

Old `as_base_exp` returned `(2/3, -1)`, but changed—`(3/2, 1)`, so numerator and denominator are not splitted, and original `1/(2/3)` is appended  to `numer`, and `denom=1`;

3. Next in `sympy/printing/latex.py`
```python
        numer, denom = fraction(expr, exact=True)  # <- numer = 1/(2/3), denom = 1 here

        if denom is S.One and Pow(1, -1, evaluate=False) not in expr.args:
            # use the original expression here, since fraction() may have
            # altered it when producing numer and denom
            tex += convert(expr)   # <- Here it goes to recursion loop trying to convert 1/(2/3)

        else:
            snumer = convert(numer)
            sdenom = convert(denom)
```

If change of function `as_base_exp` from commit `e76564ad53a297079b0c567c607e1e57a9e75de6` is reverted, print issue disappears. And the following tests fail:

```
FAILED ..\..\core\tests\test_power.py::test_pow_as_base_exp - assert (2/3, x) == (3/2, -x)
FAILED ..\..\core\tests\test_power.py::test_issue_23918 - assert (2/3, x) == (3/2, -x)
FAILED ..\..\stats\tests\test_discrete_rv.py::test_product_spaces - AssertionError: assert 'Sum(Piecewis..., (n, 1, oo))' == 'Sum(Piecewis..., (n, 1, oo))'
```

Top two are from the commit. And the bottom one:


```bash
cd sympy\sympy\stats\tests
pytest -v test_discrete_rv.py -k test_product_spaces
```

```python
def test_product_spaces():
    X1 = Geometric('X1', S.Half)
    X2 = Geometric('X2', Rational(1, 3))
    assert str(P(X1 + X2 < 3).rewrite(Sum)) == (
        "Sum(Piecewise((1/(4*2**n), n >= -1), (0, True)), (n, -oo, -1))/3")
    assert str(P(X1 + X2 > 3).rewrite(Sum)) == (
        'Sum(Piecewise((2**(X2 - n - 2)*(3/2)**(1 - X2)/6, '
        'X2 - n <= 2), (0, True)), (X2, 1, oo), (n, 1, oo))')
    assert P(Eq(X1 + X2, 3)) == Rational(1, 12)


E         - Sum(Piecewise((2**(X2 - n - 2)*(3/2)**(1 - X2)/6, X2 - n <= 2), (0, True)), (X2, 1, oo), (n, 1, oo))
E         ?                                  --    ----
E         + Sum(Piecewise((2**(X2 - n - 2)*(2/3)**(X2 - 1)/6, X2 - n <= 2), (0, True)), (X2, 1, oo), (n, 1, oo))
E         ?                                 ++       ++++
```

--- a/sympy/core/power.py
+++ b/sympy/core/power.py
@@ -820,10 +820,9 @@ class Pow(Expr):
         (1/2, 2)
 
         """
-
         b, e = self.args
-        if b.is_Rational and b.p < b.q and b.p > 0:
-            return 1/b, -e
+        if b.is_Rational and b.p == 1 and b.q != 1:
+            return Integer(b.q), -e
         return b, e
 
     def _eval_adjoint(self):

--- a/sympy/core/tests/test_power.py
+++ b/sympy/core/tests/test_power.py
@@ -282,7 +282,7 @@ def test_pow_as_base_exp():
     p = (S(3)/2)**x
     assert p.base, p.exp == p.as_base_exp() == (3*S.Half, x)
     p = (S(2)/3)**x
-    assert p.as_base_exp() == (S(3)/2, -x)
+    assert p.as_base_exp() == (S(2)/3, x)
     assert p.base, p.exp == (S(2)/3, x)
     # issue 8344:
     assert Pow(1, 2, evaluate=False).as_base_exp() == (S.One, S(2))
@@ -650,7 +650,7 @@ def test_powers_of_I():
 
 def test_issue_23918():
     b = S(2)/3
-    assert (b**x).as_base_exp() == (1/b, -x)
+    assert (b**x).as_base_exp() == (b, x)
 
 
 def test_issue_26546():

--- a/sympy/printing/tests/test_latex.py
+++ b/sympy/printing/tests/test_latex.py
@@ -259,9 +259,15 @@ def test_latex_basic():
     assert latex(Pow(Rational(1, 3), -2, evaluate=False)) == r"\frac{1}{(\frac{1}{3})^{2}}"
     assert latex(Pow(Integer(1)/100, -1, evaluate=False)) == r"\frac{1}{\frac{1}{100}}"
 
-
     p = Symbol('p', positive=True)
     assert latex(exp(-p)*log(p)) == r"e^{- p} \log{\left(p \right)}"
+
+    assert latex(Pow(Rational(2, 3), -1, evaluate=False)) == r'\frac{1}{\frac{2}{3}}'
+    assert latex(Pow(Rational(4, 3), -1, evaluate=False)) == r'\frac{1}{\frac{4}{3}}'
+    assert latex(Pow(Rational(-3, 4), -1, evaluate=False)) == r'\frac{1}{- \frac{3}{4}}'
+    assert latex(Pow(Rational(-4, 4), -1, evaluate=False)) == r'\frac{1}{-1}'
+    assert latex(Pow(Rational(1, 3), -1, evaluate=False)) == r'\frac{1}{\frac{1}{3}}'
+    assert latex(Pow(Rational(-1, 3), -1, evaluate=False)) == r'\frac{1}{- \frac{1}{3}}'
 
 
 def test_latex_builtins():

--- a/sympy/stats/drv_types.py
+++ b/sympy/stats/drv_types.py
@@ -179,7 +179,7 @@ def FlorySchulz(name, a):
     >>> X = FlorySchulz("x", a)
 
     >>> density(X)(z)
-    (5/4)**(1 - z)*z/25
+    (4/5)**(z - 1)*z/25
 
     >>> E(X)
     9
@@ -252,7 +252,7 @@ def Geometric(name, p):
     >>> X = Geometric("x", p)
 
     >>> density(X)(z)
-    (5/4)**(1 - z)/5
+    (4/5)**(z - 1)/5
 
     >>> E(X)
     5

--- a/sympy/stats/tests/test_discrete_rv.py
+++ b/sympy/stats/tests/test_discrete_rv.py
@@ -292,6 +292,6 @@ def test_product_spaces():
     assert str(P(X1 + X2 < 3).rewrite(Sum)) == (
         "Sum(Piecewise((1/(4*2**n), n >= -1), (0, True)), (n, -oo, -1))/3")
     assert str(P(X1 + X2 > 3).rewrite(Sum)) == (
-        'Sum(Piecewise((2**(X2 - n - 2)*(3/2)**(1 - X2)/6, '
+        'Sum(Piecewise((2**(X2 - n - 2)*(2/3)**(X2 - 1)/6, '
         'X2 - n <= 2), (0, True)), (X2, 1, oo), (n, 1, oo))')
     assert P(Eq(X1 + X2, 3)) == Rational(1, 12)

